### PR TITLE
If the thread does not get the lock in the short term, yield the CPU

### PR DESCRIPTION
**Pull request by stlankes**
Reduces on [RustyHermit](https://github.com/hermitcore/rusty-hermit) the amount of wasted processor cycles

### CODE REVIEW

**Comment by hkratz** [comment]
Something went wrong with your rebase. Can you drop the unrelated commits?

**Comment by stlankes** [comment]
@hkratz Yes, I dropped the unrelated commits.

File: library/std/src/sys/hermit/mutex.rs [hunk 1]
```diff
@@ -46,8 +46,17 @@ impl<T> Spinlock<T> {
     #[inline]
     fn obtain_lock(&self) {
         let ticket = self.queue.fetch_add(1, Ordering::SeqCst) + 1;
+        let mut counter: u16 = 0;
         while self.dequeue.load(Ordering::SeqCst) != ticket {
-            hint::spin_loop();
+            counter = counter + 1;
```
Comment (kennytm) [inline library/std/src/sys/hermit/mutex.rs:51]:
```suggestion
            counter += 1;
```

**Review by kennytm — APPROVED** [review]

**Pull request merged**